datastore-wrapper/python/lib/datastorewrapper.py

```diff
@@ -154,3 +154,190 @@ def files(self, repository, dataset):
+            cmd.append("--metadata-file={0}".format(metadata_file))
+
+        if publish:
+            cmd.append('--publish')
+        else:
+            cmd.append('--no-publish')
+            
+        if force:
+            cmd.append('--force')
+
+        cmd.append(repository)
+        cmd.append(dataset)
+
+        if filenames:
+            for f in filenames:
+                cmd.append(f)
+
+        out, err = self._run_with_args(cmd)
+
+        if err:
+            raise DatastoreError('Unable to create dataset "{0}": {1}'.format(dataset, err))
+
+        return True
+
+    def delete(self, repository, dataset, force = False):
+        """Deletes a dataset from a given repository
+
+        :param repository: name of the repository
+        :param dataset: name of the dataset
+        :param force: Force deleting a published dataset. Default False
+        :return: True if dataset is successfully deleted
+        """
+        self._validate_repository_and_dataset(repository, dataset)
+
+        cmd = ['delete']
+        if force:
+            cmd.append('--force-delete-published')
+
+        cmd.append(repository)
+        cmd.append(dataset)
+
+        out, err = self._run_with_args(cmd)
+
+        if err:
+            raise DatastoreError('Unable to delete: {0}'.format(err))
+
+        return True
+
+
+    def publish(self, repository, dataset):
+        """Publishes a dataset from a given repository
+
+        :param repository: name of the repository
+        :param dataset: name of the dataset
+        :return: True if dataset is successfully deleted
+        """
+        self._validate_repository_and_dataset(repository, dataset)
+
+        cmd = ['publish', repository, dataset]
+
+        out, err = self._run_with_args(cmd)
+
+        if err:
+            raise DatastoreError('Unable to publish: {0}'.format(err))
+
+        return True
+
+
+    def rebuild_file_list(self, repository, dataset):
+        """Regenerates the file list metadata on a dataset
+
+        :param repository: name of the repository
+        :param dataset: name of the dataset
+        :return: True if dataset is successfully deleted
+        """
+        self._validate_repository_and_dataset(repository, dataset)
+
+        cmd = ['rebuild-file-list', repository, dataset]
+
+        out, err = self._run_with_args(cmd)
+
+        if err:
+            raise DatastoreError('Unable to rebuild file list : {0}'.format(err))
+
+        return True
+
+    def update_metadata(self, repository, dataset, metadata = {}, metadata_file = ''):
+        """Updates the metadata of a dataset on a given repository
+
+        :param repository: name of the repository
+        :param dataset: name of the dataset
+        :param metadata: dictionary of 'description', 'author', 'author_email', 'data-type', 'version', 
+               'maintainer' and 'maintainer-email'. Optional if 'metadata_file' parameter provided
+        :param metadata_file: YAML file containing metadata
```

Comment:
The return value documentation needs to be updated in a few places. But otherwise the code looks good.